contracts/proto/faas/v1/faas.proto

```diff
@@ -176,6 +197,17 @@ message TopicTriggerContext {
   // TODO: Add the event ID to the trigger context here got transactional outbox?
 }
 
+enum NotificationType {
+  Bucket = 0;
+}
+
+message NotificationTriggerContext {
+  // The resource the message was triggered by
+  NotificationType type = 1;
+  string resource = 2;
+  map<string, string> attributes = 3; // Convention for buckets is "key" and "type"
+}
```

Comment:
If the intention here is to future proof, I would avoid including conventions specific to buckets in the base context.

It easier to use oneof in this case to get the specific type with common attributes in the base context.

e.g.

```protobuf
message BucketNotification {
  map<string, string> attributes = 2;
}

message NotificationTriggerContext {
  string source = 1;
  
  // Now we don't need to specify a type enum, the specific contents/type comes in a single field of the context.
  oneof notification {
     BucketNotification bucket = 10;
  }
}
```